### ISSUE TITLE
Default to languages indicated by browser

### ISSUE THREAD

**Issue by BalooUriza** [mastodon/mastodon#27047]
### Pitch

Detect browser languages using the Accept-Language HTTP request header, and default the UI to the prime language given and filter posts viewed by default to just the languages indicated.

### Motivation

The web browser outright tells the website it's requesting from what the language preferences of the user are.  Making posts that do not conform to those languages visible by default is usually a safe assumption.  This would prevent busy feeds (especially on small instances where only a small number of languages are relevant to the user base) much easier to navigate out of the box.

This is especially the case for users who are not logged in and thus _can't_ filter by language otherwise.

**Comment by notabene** [comment]
Good idea.

This being said, considering choice should always be left in the hands of the user, this should be bypassable through a cookie setting.

The algorithm goes something like this:

0. language is set by the interface's owner by default
1. analyse Accept-Language
2. if no Language cookie is set, set interface to most weighted Accept-language language.
3. if a cookie is set, set interface to cookie value.
4. if none of the above 1 to 4, fall back on 0.

**Comment by BalooUriza** [comment]
> This being said, considering choice should always be left in the hands of the user, this should be bypassable through a cookie setting.

Yes, exactly, this is already the case for logged in users.  Though I was also operating on the thought that setting a cookie is undesirable if it can be reasonably avoided, at least for logged-out users, and that users who need to override this can do so from their home instance.
